packages/runtime/container-runtime/src/summaryFormat.ts

```diff
@@ -74,21 +74,24 @@ export function hasIsolatedChannels(attributes: ReadFluidDataStoreAttributes): b
     return !!attributes.summaryFormatVersion && !attributes.disableIsolatedChannels;
 }
 
-export interface IContainerRuntimeMetadata extends ICreateContainerMetadata, IGCMetadata {
+// export type IReadContainerRuntimeMetadata = IContainerRuntimeMetadata | IOldContainerRuntimeMetadata;
+export interface IContainerRuntimeMetadata extends ICreateContainerMetadata, IOldContainerRuntimeMetadata, IGCMetadata {
     readonly summaryFormatVersion: 1;
     /** The last message processed at the time of summary. Only primitive property types are added to the summary. */
     readonly message: ISummaryMetadataMessage | undefined;
     /** True if channels are not isolated in .channels subtrees, otherwise isolated. */
     readonly disableIsolatedChannels?: true;
     /** The summary number for a container's summary. Incremented on summaries throughout its lifetime. */
     readonly summaryNumber?: number;
+}
+
+export interface IOldContainerRuntimeMetadata {
```

Comment:
I wonder if we need this interface and summaryCount as well. Glancing at the code, old clients should work Ok if it's not there, and I think it's Ok for "ContainerLoadStats" event to report nothing (undefined, not zero) if we do not have summaryNumber. Same for getNormalizedBlobContent - I'd just remove any notion of summaryCount

Reply:
@vladsud - You're saying that this line https://github.com/microsoft/FluidFramework/blob/4e8885617dbb1e284bc2b9f1bec2d57fb41ef18a/packages/runtime/container-runtime/src/containerRuntime.ts#L1447

should be just `loadSummaryNumber = metadata?.summaryNumber;`?

But then telemetry for old documents (created before 0.59.3000) will _never_ have `summaryNumber` set (after code is updated to handle undefined case).

How do you weigh the cost (in terms of lines of code) of supporting this logging for old documents?  I'm definitely inclined to do the work to "rescue" old documents here, but I haven't spent much time in this telemetry to understand its worth.

Reply:
Yes, the count would reset and start counting from zero. If we believe it's a big problem - we can keep it. Otherwise, I'd just it

Reply:
After some thought, I think it'd be best to remove summaryCount entirely. We could add the interface, but I don't think it's needed  to keep support for old docs that haven't been opened or read for a while. I'd argue  that this makes the code less readable as well.

Reply:
@agarwal-navin - if you ever see this comment - FYI that `summaryNumber` will be reset for old documents.  Not sure how it's used practically.  I guess we can correlate with createdByRuntimeVersion or whatever it's called to know whether the summaryNumber may have been reset?

cc @NicholasCouri and @kian-thompson in case you depend on this value in telemetry.